src/saving/serializers/RandomConverter.cs

```diff
@@ -73,11 +73,9 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
 
             return random;
         }
-        catch (ArgumentException e)
-        {
-            throw new JsonException("Can't read Random (missing property)", e);
-        }
-        catch (NullReferenceException e)
+        catch (Exception e) when (
+            e is ArgumentException ||
```

Comment:
I didn't add `ArgumentNullException` here, because rider complained that `this is always false`